climaf/classes.py

```diff
@@ -408,6 +416,20 @@ def __init__(self, **kwargs):
         self.local_copies_of_remote_files = None
         self.register()
 
+    def __eq__(self, other):
+        res = super(cdataset, self).__eq__(other)
+        if res:
+            self_kvp = copy.deepcopy(self.kvp)
+            self_kvp["model"] = self.model
+            self_kvp["frequency"] = self.frequency
+            self_kvp["alias"] = self.alias
+            other_kvp = copy.deepcopy(other.kvp)
```

Comment:
@senesis at __init__ stage, attributes model and frequency could be not defined in kvp but "*" is the default values for both of them. Alias is not defined in kvp but it is in cdataset attributes. That the reason why I did not defined the other attributes.
But the check of self.period.fx could be added too as it is also changed in cdataset attributes.
But, maybe, it would be better to directly check the dataset attributes...